viadot/flows/mediatool_to_adls.py

```diff
@@ -0,0 +1,171 @@
+import os
+from pathlib import Path
+from typing import Any, Dict, List
+
+import pendulum
+from prefect import Flow
+from prefect.backend import set_key_value
+from prefect.utilities import logging
+
+from ..task_utils import (
+    add_ingestion_metadata_task,
+    df_get_data_types_task,
+    df_map_mixed_dtypes_for_parquet,
+    df_to_csv,
+    df_to_parquet,
+    dtypes_to_json_task,
+    update_dtypes_dict,
+)
+from ..tasks import AzureDataLakeUpload, MediatoolToDF
+
+file_to_adls_task = AzureDataLakeUpload()
+json_to_adls_task = AzureDataLakeUpload()
+
+logger = logging.get_logger(__name__)
+
+
+class MediatoolToADLS(Flow):
+    def __init__(
+        self,
+        name: str = None,
+        organization_ids: List[str] = None,
+        media_entries_columns: List[str] = None,
+        mediatool_credentials: dict = None,
+        mediatool_credentials_key: str = "MEDIATOOL",
+        vault_name: str = None,
+        credentials_secret: str = None,
+        output_file_extension: str = ".parquet",
+        adls_dir_path: str = None,
+        local_file_path: str = None,
+        adls_file_name: str = None,
+        adls_sp_credentials_secret: str = None,
+        overwrite_adls: bool = False,
+        if_exists: str = "replace",
+        *args: List[Any],
+        **kwargs: Dict[str, Any],
+    ):
+        """
+        Flow for downloading data from Mediatool platform to CSV or parquet file using Mediatool API,
+        then uploading it to Azure Data Lake.
+
+        Args:
+            name (str): The name of the flow.
```

Comment:
You can specify if name is required or not - in line 30 you are saying that default is None and is not required.